spec/controllers/users/registrations_controller_spec.rb

```diff
@@ -0,0 +1,152 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'support/shared/registration_controller_shared'
+
+RSpec.describe Users::RegistrationsController, type: :controller do
+  describe 'route' do
+    it { is_expected.to_not route(:get, '/api/users/signup').to(action: :index, format: :json) }
+    it { is_expected.to_not route(:patch, '/api/users/signup').to(action: :show, format: :json) }
+    it { is_expected.to_not route(:get, '/api/users/signup/sign_up').to(action: :new, format: :json) }
+    it { is_expected.to_not route(:get, '/api/users/signup/edit').to(action: :edit, format: :json) }
+    it { is_expected.to_not route(:delete, '/api/users/signup').to(action: :destroy, format: :json) }
+    it { is_expected.to route(:post, '/api/users/signup').to(action: :create, format: :json) }
+    it { is_expected.to route(:put, '/api/users/signup').to(action: :update, format: :json) }
+    it { is_expected.to route(:patch, '/api/users/signup').to(action: :update, format: :json) }
+  end
+
+  describe 'POST#create' do
+    before(:each) { @request.env['devise.mapping'] = Devise.mappings[:user] }
+
+    let(:send_request) { post :create, params: params, as: :json }
+    let(:params) do
+      {
+        user: {
+          email: email,
+          phone: phone
+        }
+      }
+    end
+    let(:email) { Faker::Internet.email }
+    let(:phone) { Faker::Base.numerify('###-###-####') }
+
+    it 'returns json' do
+      send_request
+      expect(response.content_type).to include('application/json')
+    end
+
+    context 'with user email' do
+      let(:phone) { nil }
+
+      it 'creates user' do
+        expect { send_request }.to change(User, :count)
+      end
+      it 'gets 201 code' do
+        send_request
+        expect(response).to have_http_status(:created)
+      end
+    end
+    context 'with user phone' do
+      let(:email) { nil }
+
+      it 'creates user' do
+        expect { send_request }.to change(User, :count)
+      end
+      it 'gets 201 code' do
+        send_request
+        expect(response).to have_http_status(:created)
+      end
+    end
+    context 'without user email and phone' do
+      let(:email) { nil }
+      let(:phone) { nil }
+
+      it 'doesn\'t create user' do
+        expect { send_request }.to_not change(User, :count)
+      end
+      it 'gets 422 code' do
+        send_request
+        expect(response).to have_http_status(:unprocessable_entity)
+      end
+    end
+  end
+
+  describe 'POST#update' do
+    before(:each) { @request.env['devise.mapping'] = Devise.mappings[:user] }
+
+    let(:create_request) do
+      post :create, params: { user: params[:user].slice(:email, :phone) }, as: :json
+      request.headers['token'] = JSON.parse(response.body)['data']['token']
+    end
```

Comment:
why can't you just create user using factory_bot?

Reply:
I did it because I need token for update request, and I can`t get it from creating through factory. But I can get token from create action. In fact that was not needed, because I can get token with JsonWebToken.encode

Reply:
you can also stub it, for example

```ruby
class ApplcationController
  def current_user
    User.find ...
  end
end

#rspec
let(:user) { create(:user) } 
it do
  allow(controller).to receive(:current_user).and_return(user)
  #...
end
```

Reply:
this is still not fixed